src/System Application/App/Azure Key Vault/src/AzureKeyVaultImpl.Codeunit.al

```diff
@@ -23,44 +23,24 @@ codeunit 2202 "Azure Key Vault Impl."
         NavAzureKeyVaultClient: DotNet AzureKeyVaultClientHelper;
         [NonDebuggable]
         AzureKeyVaultSecretProvider: DotNet IAzureKeyVaultSecretProvider;
-        SecretNotFoundErr: Label '%1 is not an application secret.', Comment = '%1 = Secret Name.';
         [NonDebuggable]
         CachedSecretsDictionary: Dictionary of [Text, Text];
         [NonDebuggable]
         CachedCertificatesDictionary: Dictionary of [Text, Text];
-        AllowedApplicationSecretsSecretNameTxt: Label 'AllowedApplicationSecrets', Locked = true;
-        [NonDebuggable]
-        AllowedSecretNamesList: List of [Text];
         IsKeyVaultClientInitialized: Boolean;
-        NoSecretsErr: Label 'The key vault did not have any secrets that are allowed to be fetched.';
-        AllowedApplicationSecretsSecretNotFetchedMsg: Label 'The list of allowed secret names could not be fetched.', Locked = true;
         AzureKeyVaultTxt: Label 'Azure Key Vault', Locked = true;
-        InitializeAllowedSecretNamesErr: Label 'Initialization of allowed secret names failed.';
         CertificateInfoTxt: Label 'Successfully constructed certificate from secret %1. Certificate thumbprint %2', Locked = true;
 
     [NonDebuggable]
     procedure GetAzureKeyVaultSecret(SecretName: Text; var Secret: Text)
     begin
         // Gets the secret as a Text from the key vault, given a SecretName.
-
-        if not InitializeAllowedSecretNames() then
-            Error(InitializeAllowedSecretNamesErr);
-
-        if not IsSecretNameAllowed(SecretName) then
-            Error(SecretNotFoundErr, SecretName);
-
         Secret := GetSecretFromClient(SecretName);
```

Comment:
Clean up this function, as the only thing it does is call another function.

Reply:
Yea, but this is called twice from a method that returns text and secret text